### ISSUE TITLE
Can't terminate wayfire via kill -15 with glib-main-loop enabled

### ISSUE THREAD

**Issue by ssfdust** [WayfireWM/wayfire#2386]
**Describe the bug**
With glib-main-loop enabled, we can't kill wayfire. Wayfire is still running and doesn't print any shutdown information.

**To Reproduce**
1. Enable plugin glib-main-loop
2. kill -9 <pid of wayfire>
3. It still runs

**Expected behavior**
Wayfire shutdowns

**Screenshots or stacktrace**
If it is a crash, attach the backtrace (or the whole log file), Wayfire will print it in the end of the log file or stdout.
Backtrace with address sanitizer enabled (if possible):

**Wayfire version**
latest git

**Comment by ammen99** [comment]
Makes sense, glib-main-loop replaces the wayland event loop. So it will have to override the signal handlers to stop its own event loop (wayfire-core uses core::shutdown() to shut down the wayland event loop, which is only part of the story). Alternatively, glib-main-loop should listen to the shutdown signal.
